File: examples/csharp/avalonia/NotesShowcase/ViewModels/ActionVM.cs
using System.Windows.Input;

namespace NotesShowcase.ViewModels;

/// <summary>
/// Pure presentation record for a single capability-derived action.
///
/// Used by <see cref="CapabilityActionsVM"/> to project a focused VM's
/// capability surface into a flat list of (label, command) tuples for the view.
/// See plan §3.a.5 and spec §14.4 (capability dispatch).
/// </summary>
public sealed record ActionVM(string Label, ICommand Command);

File: examples/typescript/react/notes-showcase/src/viewmodels/actionVM.ts
/**
 * ActionVM — pure presentation record for one capability-derived action.
 *
 * Used by CapabilityActionsVM to project a focused VM's capability surface
 * into a flat list of (label, command) tuples for the view layer (Phase 5.c).
 *
 * See spec §14.4 (capability dispatch) and the C# `ActionVM.cs` parity record.
 */
import type { ICommand } from "vmx";

export interface ActionVM {
  readonly label: string;
  readonly command: ICommand;
}

/** Helper factory mirroring the C# / Python `ActionVM(label, command)` constructor. */
export function makeActionVM(label: string, command: ICommand): ActionVM {
  return { label, command };
}
